File: rx-codelab/src/main/java/com/jraska/rx/codelab/http/IpViewModel.java
package com.jraska.rx.codelab.http;

import com.jraska.rx.codelab.SchedulerProvider;
import io.reactivex.Observable;

public class IpViewModel {
  private final HttpBinApi httpBinApi;
  private final SchedulerProvider schedulerProvider;

  public IpViewModel(HttpBinApi httpBinApi, SchedulerProvider schedulerProvider) {
    this.httpBinApi = httpBinApi;
    this.schedulerProvider = schedulerProvider;
  }

  public Observable<String> ip() {
    return httpBinApi.getRequest()
      .subscribeOn(schedulerProvider.getIo())
      .observeOn(schedulerProvider.getMain())
      .map(requestInfo -> requestInfo.origin);
  }
}

File: rx-codelab/src/main/java/com/jraska/rx/codelab/http/IpViewModel.kt
package com.jraska.rx.codelab.http

import com.jraska.rx.codelab.SchedulerProvider
import io.reactivex.Observable

class IpViewModel(private val httpBinApi: HttpBinApi, private val schedulerProvider: SchedulerProvider) {

  fun ip(): Observable<String> {
    return httpBinApi.getRequest()
      .subscribeOn(schedulerProvider.io)
      .observeOn(schedulerProvider.main)
      .map { requestInfo -> requestInfo.origin }
  }
}
